osu.Game/Rulesets/Objects/Legacy/ConvertHitObjectParser.cs

```diff
@@ -32,6 +32,8 @@ public abstract class ConvertHitObjectParser : HitObjectParser
 
         protected bool FirstObject { get; private set; } = true;
 
+        protected bool AllowLayeredHitSounds { get; set; } = true;
```

Comment:
This cannot exist in the parser - it should only exist within the legacy skin itself.

The mania override can be done inside `LegacyManiaSkinTransformer`.

That is to say, layered hitsounds should _always_ be added.